### ISSUE TITLE
Issues with Most Popular Pages module responsiveness on 599px to 668px viewports

### ISSUE THREAD

**Issue by mohitwp** [google/site-kit-wp#7563]
## Bug Description

Analytics most popular pages widget is not properly appearing on viewport having width between 599px to 668px.

## Steps to reproduce

1. Set up site kit with Analytics.
2. Go to main dashboard.
3. Set viewport between 599px to 668px using dev tool.
4. See issue.

## Screenshots

<Details>
<Summary>Screenshots</Summary>

![image](https://github.com/google/site-kit-wp/assets/94359491/747eea53-1e4d-4324-bf82-ee7ae1de8e9c)

![image](https://github.com/google/site-kit-wp/assets/94359491/db922390-039a-4862-94ef-8e5c51c920e9)

![image](https://github.com/google/site-kit-wp/assets/94359491/05e824a6-f0bc-44a7-848a-2ca15cf22229)
</Details>

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* For the Top content over 28 days table, on mobile viewports where the table headings start to overlap with each other, the table should be converted to a two column layout with tabs that as per this [Figma mock](https://www.figma.com/design/nHnGFJizcJCyTVNZwLE8Vz/Site-Kit-Misc?node-id=54-946&t=5gdpPGM2gJb7bTLc-0).

<Details>
<Summary>Screenshot</Summary>
<img width="1021" alt="Screenshot 2024-07-01 at 15 42 04" src="https://github.com/google/site-kit-wp/assets/1721319/0917e5b8-b47a-4b92-9c85-d1a73d1edb7d">
</Details>

## Implementation Brief

**Report Table Tabbed View**
* [ ] Update `assets/js/components/ReportTable.js`
   * [ ] Add a new boolean prop `tabbedLayout`, if this prop is true, render the following alternate component structure:
       * [ ] Use the `TabBar` and `Tab` components to render a new tab bar above the report table, mapping over the `columns` props to get the tab titles and keys.
       * [ ] Store the currently selected tab in component state.
       * [ ] Update the table to render only the column selected in the active tab.
 * [ ] Update `assets/sass/components/global/_googlesitekit-table.scss`, matching the [Figma mock](https://www.figma.com/design/nHnGFJizcJCyTVNZwLE8Vz/Site-Kit-Misc?node-id=54-946&t=5gdpPGM2gJb7bTLc-0) for this tabbed version of the report table.

**Most Popular Pages module**
* [ ] Update `assets/js/modules/analytics-4/components/module/ModulePopularPagesWidgetGA4/index.js`:
   * [ ] Use the `useBreakpoint` hook to get the current breakpoint.
   * [ ] Pass the true to the `tabbedLayout` prop of the ReportTable in this component if the breakpoint is `BREAKPOINT_TABLET` or `BREAKPOINT_MOBILE`.

### Test Coverage

* In `assets/js/components/ReportTable.stories.js`, create a new VRT for the tabbed version of this component.
* Confirm there are no regressions to other key widgets that share this component.

## QA Brief

* Site up Site Kit with Analytics connected.
* Verify the "top content" Site Kit dashboard widget is rendered in a tabbed view at mobile viewports as per the AC.
* Additionally, check the page title font is also bold when viewed in the desktop viewport, and in the WP dashboard "top content" widget as discussed in [Slack](https://fueled10up.slack.com/archives/C01CA828ZPZ/p1732616012510899).

Please note, the widget header has been intentionally left the same to be consistent with other widgets on the dashboard:

![Image](https://github.com/user-attachments/assets/779aa03a-ff25-49fb-a374-7cadb79ddb06)

![Image](https://github.com/user-attachments/assets/ee7cdf9b-edc8-4595-9ed6-d914287845bb)

## Changelog entry

* Update layout of Top Content widget in smaller viewports to use a tab per metric column.

**Comment by jimmymadon** [comment]
@sigal-teller @aaemnnosttv This seems to be an issue only for a very short viewport range between 600px and 670px. I suggest simply wrapping the words, ideally with CSS hyphens. Does this look ok?

<img width="620" alt="Screenshot 2024-05-31 at 02 21 12" src="https://github.com/google/site-kit-wp/assets/1721319/1d08cc54-c280-455b-b34d-f69d198aa864">

**Comment by techanvil** [comment]
Hey @jimmymadon, it's worth noting we have a similar issue in the backlog, https://github.com/google/site-kit-wp/issues/5561, where we looked into hyphenation, among other things - see https://github.com/google/site-kit-wp/issues/5561#issuecomment-1191770816.

However as discussed there are issues with this approach; the direction from Seb toward the end of the issue was to simply change the font size.

**Comment by jimmymadon** [comment]
@sigal-teller Could you please review the solutions proposed by @techanvil and @aaemnnosttv from this comment [thread](https://github.com/google/site-kit-wp/issues/5561#issuecomment-1191770816).

For me personally, the hyphens work nicely. But as Tom and Evan say, using icons are also a good option which will keep the height consistent too.

**Comment by jimmymadon** [comment]
@sigal-teller You suggested adding horizontal scroll which we have introduced for the Audience Segmentation widget. In experimenting with the horizontal scroll, I realised we could further reduce the width of column 1 with minimal code change and effort.

<img width="558" alt="Screenshot 2024-06-19 at 11 50 19" src="https://github.com/google/site-kit-wp/assets/1721319/765b68fc-b6b6-4d41-949c-5dbeaed0ae27">

**Comment by sigal-teller** [comment]
Thank you @jimmymadon for these explorations. I think it doesn't look so great especially with long URLs and page names. Does using horizontal scrolling here adds a lot of complexity?

**Comment by jimmymadon** [comment]
@sigal-teller Thank you - horizontal scrolling shouldn't be a big issue. I'll add that as the AC here, move this along and we can see the estimate that comes out from the IB.

**Comment by tofumatt** [comment]
Having the remaining columns be horizontally scrollable while the first column remains fixed in place is likely significantly more complicated/effort than simply scrolling the entire table, which is probably an easier solution. 🤔

I can kind of understand keeping the first column around for the context, but it will also make it so the user needs to horizontally scroll when using touch gestures ONLY after the first column. That's not intuitive or normal scroll behaviour, especially when there's no design here for a visual indicator that there's more content.

I think we should be scrolling the entire table/widget contents, not select columns. It presents significant accessibility/usability issues, especially on touchscreen devices, where these viewports are likely to be encountered.

**Comment by aaemnnosttv** [comment]
FWIW, these originally were designed as horizontally scrollable in smaller viewports and this was later changed for some reason I can't recall.

On a related note, we should scroll the top navigation if it overflows (a separate issue) as this is a common pattern. 
<img width="740" alt="image" src="https://github.com/google/site-kit-wp/assets/1621608/7f0216a6-4ee2-44a4-8596-66953d169ae1">

**Comment by sigal-teller** [comment]
@tofumatt I agree that it's more complicated to scroll only some of the columns in several aspects. The question is if it makes sense to see the numbers in the table without the page names, and if the user will need to scroll back and forth again and again. 

@jimmymadon I have another idea here that from a UX perspective is much better but might require more effort. Instead of having 1 table with scrolling we can have 4 tables that will include the 1st column and one more column, and we'll add tabs above the table to switch between them: "Pageviews", "Sessions" etc. This will be vwery similar to the pattern we introduced in the Audience segmentation [UX](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=1936-22975&t=VdSNtEpSwLCnUWXu-1) for multiple groups in narrow viewports. I also found this pattern in SC (attaching a screenshot)
![IMG_9912](https://github.com/google/site-kit-wp/assets/69191559/c19a23f6-8e3a-4fef-8fa5-3b074cc6a775)

@aaemnnosttv As for the top navigation, we should definitely add a horizontal scroll instead of breaking into 2 lines. I suggested it a while ago when I worked on the header improvements which were paused, but I'm using it in all my designs since then. you can see it [here](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=1655-35687&t=VdSNtEpSwLCnUWXu-1) for example.

**Comment by jimmymadon** [comment]
@sigal-teller 

>This will be vwery similar to the pattern we introduced in the Audience segmentation [UX](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=1936-22975&t=VdSNtEpSwLCnUWXu-1) for multiple groups in narrow viewports.

If @tofumatt or another AC reviewer (@eugene-manuilov?) can review this solution, it might be better to have a formal design of this in Figma which we can use for all multiple column tables with a wide first column.

**Comment by sigal-teller** [comment]
@jimmymadon I added an example [here](https://www.figma.com/design/nHnGFJizcJCyTVNZwLE8Vz/Site-Kit-Misc?node-id=54-946&t=mWqNvzaFIKii08Lk-1) for the Figma design I referred to in my previous comment. LMK if that can work.

**Comment by tofumatt** [comment]
That tabbed solution seems great, I think that's a more intuitive one than scrolling; it'd be more straightforward to implement and I think is flexible enough to even allow for more tabs than fit in the widget, by scrolling only the tabs.

Looks good to me, moving to IB 👍🏻

**Comment by eugene-manuilov** [comment]
@benbowler, instead of making `ReportTable` breakpoint aware, we should read breakpoint data in a component that requires an alternative layout for the table based on the current breakpoint. In other words, we should update the report table to have an alternative layout and the component that renders that table to read the current breakpoint and to use an alternative layout if the breakpoint is tablet or mobile.

**Comment by benbowler** [comment]
@eugene-manuilov sounds good, I've updated the IB.

**Comment by wpdarren** [comment]
### QA Update: ⚠ 

@benbowler @techanvil I am confused by the solution we have gone with on this one.   I was expecting to see my test site look like the figma designs with the tabbed view.  I am on the develop branch and using data from oi.ie.  While this is a new site, I did clear cache and refresh the develop branch a few times. 

This is what I see at 599 px
![Image](https://github.com/user-attachments/assets/c6ad7871-0e6f-4614-a053-9d8cf0b56121)

This is what I see at 600 px
![Image](https://github.com/user-attachments/assets/4cbdeb9f-bf74-409e-b80f-626cbae5a949)

The overlapping continues throughout to 668 px.

Am I missing something here?

**Comment by techanvil** [comment]
Thanks for flagging this @wpdarren, it turns out the PR was approved but not merged although the rest of the process was completed with the issue moved to QA. This was clearly an oversight so I've gone ahead and merged the PR.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

Love the solution for this on mobile!  Looks great. 

- For the Top content over 28 days table, on mobile viewports where the table headings start to overlap with each other, the table is converted to a two column layout with tabs that as per this [Figma mock](https://www.figma.com/design/nHnGFJizcJCyTVNZwLE8Vz/Site-Kit-Misc?node-id=54-946&t=5gdpPGM2gJb7bTLc-0).
- I checked the page title font is also bold when viewed in the desktop viewport, and in the WP dashboard "top content" widget as discussed in [Slack](https://fueled10up.slack.com/archives/C01CA828ZPZ/p1732616012510899).

https://github.com/user-attachments/assets/edffe7a6-db34-4c1c-8aef-8a3fbdff790a

<details>
 <summary>Screenshots</summary>

![Image](https://github.com/user-attachments/assets/04f345d0-2efc-4b7f-905e-0bbb46c71903)
![Image](https://github.com/user-attachments/assets/8319749c-6ecc-43a9-82f1-3098ad0497a3)


</details>
